sdk/provisioning/Azure.Provisioning/tests/SampleTests.cs

```diff
@@ -0,0 +1,327 @@
+                                {
+                                    // TODO: Decide if we want to invest in a less janky way to use expressions as keys
+                                    { BicepFunction.Interpolate($"{mi.Id}").Compile().ToString(), new UserAssignedIdentityDetails() }
+                                }
+                            }
+                    };
+                acr.AssignRole(ContainerRegistryBuiltInRole.AcrPull, mi);
+                OperationalInsightsWorkspace law =
+                    new(nameof(law))
+                    {
+                        Location = location,
+                        Sku = new OperationalInsightsWorkspaceSku() { Name = OperationalInsightsWorkspaceSkuName.PerGB2018 },
+                        Tags = tags,
+                    };
+                ContainerAppManagedEnvironment cae =
+                    new(nameof(cae))
+                    {
+                        Location = location,
+                        WorkloadProfiles =
+                        {
+                            new ContainerAppWorkloadProfile()
+                            {
+                                Name = "consumption",
+                                WorkloadProfileType = "Consumption"
+                            }
+                        },
+                        AppLogsConfiguration =
+                            new ContainerAppLogsConfiguration()
+                            {
+                                Destination = "log-analytics",
+                                LogAnalyticsConfiguration = new ContainerAppLogAnalyticsConfiguration()
+                                {
+                                    CustomerId = law.CustomerId,
+                                    SharedKey = law.GetKeys().PrimarySharedKey,
+                                }
+                            },
+                        Tags = tags,
+                    };
+                cae.AssignRole(AppContainersBuiltInRole.Contributor, mi);
+
+                // Hack in the Aspire Dashboard as a literal since there's no
+                // management plane library support for dotNetComponents yet
+                BicepLiteral aspireDashboard =
+                    new("aspireDashboard",
+                        new ResourceStatement(
+                            "aspireDashboard",
+                            new StringLiteral("Microsoft.App/managedEnvironments/dotNetComponents@2024-02-02-preview"),
+                            new ObjectExpression(
+                                new PropertyExpression("name", "aspire-dashboard"),
+                                new PropertyExpression("parent", new IdentifierExpression(cae.ResourceName)),
+                                new PropertyExpression("properties",
+                                    new ObjectExpression(
+                                        new PropertyExpression("componentType", new StringLiteral("AspireDashboard")))))));
```

Comment:
Or hell, even add whole resources we don't technically support yet.

_(I will actually fix bugs when you find them, but the great thing is you're not blocked on my turnaround time.)_